workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/views/static_report.py

```diff
@@ -300,50 +300,37 @@ def histogram_plots(diagnostics: Iterable[xr.Dataset]) -> HVPlot:
 # New plotting routines can be registered here.
 @metrics_plot_manager.register
 def time_mean_bias_metrics(metrics: RunMetrics) -> hv.HoloMap:
-    return generic_metric_plot(metrics, "time_and_global_mean_bias")
+    return metric_type_table(metrics, "time_and_global_mean_bias")
 
 
 @metrics_plot_manager.register
 def rmse_time_mean_metrics(metrics: RunMetrics) -> hv.HoloMap:
-    return generic_metric_plot(metrics, "rmse_of_time_mean")
+    return metric_type_table(metrics, "rmse_of_time_mean")
 
 
 @metrics_plot_manager.register
 def rmse_3day_metrics(metrics: RunMetrics) -> hv.HoloMap:
-    return generic_metric_plot(metrics, "rmse_3day")
+    return metric_type_table(metrics, "rmse_3day")
 
 
 @metrics_plot_manager.register
 def drift_metrics(metrics: RunMetrics) -> hv.HoloMap:
-    return generic_metric_plot(metrics, "drift")
-
-
-def generic_metric_plot(metrics: RunMetrics, metric_type: str) -> hv.HoloMap:
-    hmap = hv.HoloMap(kdims=["metric"])
-    bar_opts = dict(norm=dict(framewise=True), plot=dict(width=600))
-    variables = metrics.get_metric_variables(metric_type)
-    for varname in variables:
-        s = metrics.get_metric_all_runs(metric_type, varname)
-        bars = hv.Bars((s.run, s.value), hv.Dimension("Run"), s.units.iloc[0])
-        hmap[metrics.metric_name(metric_type, varname)] = bars
-    if len(variables) > 0:
-        return HVPlot(hmap.opts(**bar_opts))
-
-
-def get_metrics_table(
-    metrics: RunMetrics, metric_types: Sequence[str], variable_names: Sequence[str]
-) -> Mapping[str, Mapping[str, float]]:
-    """Structure a set of metrics in format suitable for reports.create_html"""
-    table = {}
-    for metric_type in metric_types:
-        for name in variable_names:
-            units = metrics.get_metric_units(metric_type, name, metrics.runs[0])
-            type_label = f"{name} {metric_type} [{units}]"
-            table[type_label] = {
-                run: f"{metrics.get_metric_value(metric_type, name, run):.2f}"
-                for run in metrics.runs
-            }
-    return table
+    return metric_type_table(metrics, "drift")
+
+
+def metric_type_table(metrics: RunMetrics, metric_type: str) -> hv.HoloMap:
+    """Return HTML for metrics table of all variables of metric_type"""
+    df = metrics.get_metric_type_table(metric_type).transpose()
+    header = f"<h3>{metric_type}</h3>"
+    return RawHTML(header + df.to_html(justify="center", float_format="{:.2f}".format))
```

Comment:
Would it make sense to make a class `Pandas` in report that hides this conversion logic?

Reply:
I'm going to leave the conversion here to avoid adding the `pandas` dependency to the report package.

Reply:
ok.